src/server/game/Entities/Creature/Creature.h

```diff
@@ -478,6 +478,7 @@ class Creature : public Unit, public GridObject<Creature>, public MovableMapObje
     ///// TODO RENAME THIS!!!!!
     bool isCanInteractWithBattleMaster(Player* player, bool msg) const;
     bool isCanTrainingAndResetTalentsOf(Player* player) const;
+    bool IsValidTrainerForPlayer(Player* player, uint32* npcFlags = nullptr) const;
```

Comment:
unless it has side effects, please always mark your methods with `[[nodiscard]]`